testapi.pm

```diff
@@ -822,6 +825,77 @@ sub eject_cd {
     query_isotovideo('backend_eject_cd');
 }
 
+=head2 save_memory_dump
+
+  save_memory_dump($filename);
+
+Saves the SUT memory state using C<$filename> as base for the memory dump filename, 
+the default will be the current test's name. 
+
+I<Currently only qemu backend is supported.>
+
+=cut
+
+sub save_memory_dump {
+
+    my $filename = shift || ref($autotest::current_test);
+    die "Method should be called within a post_fail_hook" unless ((caller(1))[3]) !~ /post_fail_hook/;
+
+
+    bmwqemu::log_call();
+    bmwqemu::diag("Trying to save machine state");
+    query_isotovideo('backend_save_memory_dump', {filename => $filename});
+
+}
+
+=head2 save_storage_drives
+
+  save_storage_drives($filename);
+
+Saves all of the SUT drives using C<$filename> as part of the final filename,
+the default will be the current test's name. The disk number will be always present.
+
+I<Currently only qemu backend is supported.>
+
+=cut
+
+sub save_storage_drives {
+
+    my $filename = shift || ref($autotest::current_test);
+    die "Method should be called within a post_fail_hook" unless ((caller(1))[3]) !~ /post_fail_hook/;
+
+    bmwqemu::log_call();
+    bmwqemu::diag("Trying to save machine drives");
+    bmwqemu::load_vars();
+
+    #TODO: right now, we're saving all the disks
+    #      sometimes we might not want to. This should be improved.
+    if (my $nd = $bmwqemu::vars{NUMDISKS}) {
+        for my $i (1 .. $nd) {
+            query_isotovideo('backend_save_storage_drives', {disk => $i, filename => $filename});
+        }
+    }
+}
+
+=head2 freeze_vm
+
+  freeze_vm;
+
+If the backend supports it, freeze the vm. This will allow the vm to be paused/frozen
+within the test, but only from the post_fail_hook. So that memory and disk dumps
+can be extracted without any risk of data changing. 
+
+It is assumed that the test writer will call freeze_vm before getting the memory or disk dumps.
```

Comment:
Doc was changed to explain why freeze_vm should be called when doing both, disk and memory dump.